crates/iota/src/client_commands.rs

```diff
@@ -3047,3 +3051,35 @@ fn parse_emit_option(s: &str) -> Result<HashSet<EmitOption>, String> {
 
     Ok(options)
 }
+
+/// Warn the user if the CLI does not match the version of current on-chain
+/// protocol.
+async fn check_protocol_version_and_warn(client: &IotaClient) -> Result<(), anyhow::Error> {
+    let on_chain_protocol_version = client
+        .read_api()
+        .get_protocol_config(None)
+        .await?
+        .protocol_version
+        .as_u64();
+
+    let cli_protocol_version = ProtocolVersion::MAX.as_u64();
+    if cli_protocol_version != on_chain_protocol_version {
+        let warning_msg = format!(
+            "[warning] The CLI's build protocol version is {cli_protocol_version}, but the current \
+            on-chain protocol version is {on_chain_protocol_version}."
```

Comment:
I like how there was a notion of "active network" in the upstream message, it's very explicit, there is only one active network. Here IMO "current" is a bit vague, not sure what we are talking about.

Reply:
@Alex6323

Reply:
Changed in https://github.com/iotaledger/iota/pull/5042/commits/bce0eb0591b9f7098f7c0e76a700c2fa9a768f29